index.js

```diff
@@ -36,13 +78,32 @@ module.exports = function (opts = {}) {
     }
 
     app.authboot = {};
+
+    const concurrency = opts.maxAuthConcurrency
+      || options.maxAuthConcurrency
+      || app.config.get('auth:maxAuthConcurrency')
+      || 3;
+
+    const algorithm = opts.algorithm
+      || options.algorithm
+      || app.config.get('auth:algorithm')
+      || 'sha256';
+
+    let checker;
     const lookup = app.authboot.lookup = (lookupOpt || function ({ name, password }, callback) {
-      const hash = users.get(name);
-      if (!hash) {
+      const hashBuffer = users.get(name);
+      if (!hashBuffer) {
         debug(`unknown username ${name}`);
         return callback(null, false);
       }
-      bcrypt.compare(password, hash, callback);
+      checker = checker || coalesce(concurrency, (password, done) => {
+        const passwordHash = crypto.createHash(algorithm);
+        passwordHash.update(password);
+        const digest = passwordHash.digest();
+        const equal = crypto.timingSafeEqual(digest, hashBuffer);
```

Comment:
should hasBuffer be getting passed in here since it's request specific?

Reply:
*username specific

Reply:
not sure what you mean, we have to check the provided username hashed value to compare against the request's guess at the password

Reply:
Sorry should have been more descriptive.

I'm worried about the closure over `hashBuffer` here since it's generated from the username on the first pass time through the lookup function.  `checker` is only ever generated once on first invocation, which means hashBuffer is only ever the first user that this was called with and not the active user.  Do we have some test cases to show that this works with multiple valid username:password pairs?

Reply:
ah, yes this is a bug and a missing test